### PR TITLE
Support NACK without distinct PayloadTypes

### PR DESCRIPTION
**Pull request by Sean-Der**
If not RTX payload type is set use the same payload type as
the sender.

### CODE REVIEW

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190?src=pr&el=h1) Report
> Merging [#190](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190?src=pr&el=desc) into [master](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/commit/a805037cac46791915bde09f2365c0559f6ae917&el=desc) will **decrease** coverage by `0.03%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #190      +/-   ##
==========================================
- Coverage   86.90%   86.86%   -0.04%     
==========================================
  Files          33       33              
  Lines        7644     7638       -6     
==========================================
- Hits         6643     6635       -8     
- Misses       1001     1003       +2     
```


| [Impacted Files](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/source/Ice/IceAgentStateMachine.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9JY2UvSWNlQWdlbnRTdGF0ZU1hY2hpbmUuYw==) | `84.34% <0.00%> (-0.22%)` | :arrow_down: |
| [src/source/PeerConnection/Rtp.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9QZWVyQ29ubmVjdGlvbi9SdHAuYw==) | `77.53% <0.00%> (+2.17%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190?src=pr&el=footer). Last update [2dce050...8ba3d63](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: src/source/PeerConnection/Retransimitter.c [hunk 1]
```diff
@@ -95,9 +96,16 @@ STATUS resendPacketOnNack(PRtcpPacket pRtcpPacket, PKvsPeerConnection pKvsPeerCo
         CHK(retStatus == STATUS_SUCCESS, retStatus);
 
         if (pRtpPacket != NULL) {
+            if (pSenderTranceiver->sender.rtxSequenceNumber == pSenderTranceiver->sender.rtxPayloadType) {
```
Comment (MushMal) [inline src/source/PeerConnection/Retransimitter.c]: Looks good but do you mind adding a comment of what it means?

File: src/source/PeerConnection/Retransimitter.c [hunk 2]
```diff
@@ -59,24 +59,20 @@ STATUS resendPacketOnNack(PRtcpPacket pRtcpPacket, PKvsPeerConnection pKvsPeerCo
     CHK_STATUS(rtcpNackListGet(pRtcpPacket->payload, pRtcpPacket->payloadLength, &senderSsrc, &receiverSsrc, NULL, &filledLen));
 
     CHK_STATUS(doubleListGetHeadNode(pKvsPeerConnection->pTransceievers, &pCurNode));
-    while(pCurNode != NULL && (pReceiverTransceiver == NULL || pSenderTranceiver == NULL)) {
+    while(pCurNode != NULL && (pSenderTranceiver == NULL)) {
```
Comment (MushMal) [inline src/source/PeerConnection/Retransimitter.c]: no need for parenthesis around (pSenderTranseiver == NULL) as you've removed the ||

File: src/source/PeerConnection/Retransimitter.c [hunk 3]
```diff
@@ -95,17 +91,19 @@ STATUS resendPacketOnNack(PRtcpPacket pRtcpPacket, PKvsPeerConnection pKvsPeerCo
         CHK(retStatus == STATUS_SUCCESS, retStatus);
 
         if (pRtpPacket != NULL) {
-            CHK_STATUS(constructRetransmitRtpPacketFromBytes(pRtpPacket->pRawPacket, pRtpPacket->rawPacketLength,
-                    pSenderTranceiver->sender.rtxSequenceNumber, pSenderTranceiver->sender.rtxPayloadType, pSenderTranceiver->sender.rtxSsrc, &pRtxRtpPacket));
-            pSenderTranceiver->sender.rtxSequenceNumber++;
+            if (pSenderTranceiver->sender.payloadType == pSenderTranceiver->sender.rtxPayloadType) {
```
Comment (MushMal) [inline src/source/PeerConnection/Retransimitter.c]: We should really start putting proper comments. Not part of this CR obviously, but I see no comments in major logic aspect while I see comments like // resendPacket which tells me nothing as a reader.

File: src/source/PeerConnection/Rtp.c [hunk 4]
```diff
@@ -181,19 +181,27 @@ STATUS writeFrame(PRtcRtpTransceiver pRtcRtpTransceiver, PFrame pFrame)
...
+            pRtpPacket->rawPacketLength = packetLen;
+            CHK_STATUS(rtpRollingBufferAddRtpPacket(pKvsRtpTransceiver->sender.packetBuffer, pRtpPacket));
+        }
+
+        signedPacketLen = packetLen;
         rawPacket = MEMREALLOC(rawPacket, packetLen + 10); // For SRTP authentication tag
-        pRtpPacket->pRawPacket = rawPacket;
-        pRtpPacket->rawPacketLength = packetLen;
-        if (pKvsRtpTransceiver->sender.packetBuffer != NULL) {
+        CHK_STATUS(encryptRtpPacket(pKvsPeerConnection->pSrtpSession, rawPacket, &signedPacketLen));
```
Comment (MushMal) [inline src/source/PeerConnection/Rtp.c]: Shouldn't these be in an else clause of !bufferAfterEncrypt? I am just not following the logic perhaps.

File: src/source/PeerConnection/Rtp.c [hunk 5]
```diff
@@ -181,19 +181,27 @@ STATUS writeFrame(PRtcRtpTransceiver pRtcRtpTransceiver, PFrame pFrame)
...
+        CHK_STATUS(iceAgentSendPacket(pKvsPeerConnection->pIceAgent, rawPacket, signedPacketLen));
+
+        if (bufferAfterEncrypt) {
+            pRtpPacket->pRawPacket = rawPacket;
+            pRtpPacket->rawPacketLength = signedPacketLen;
```
Comment (MushMal) [inline src/source/PeerConnection/Rtp.c]: Winder if the logic should be to simply check if the packet needs encryption in which case re-alloc and encrypt and set the new size. Set the size in pRtpPacket bookkeeping, add to rolling buffer and then ice agent send packet?

File: src/source/PeerConnection/Rtp.c (hunk 4 above)
Comment (Sean-Der) [reply src/source/PeerConnection/Rtp.c]: We always need to encrypt+send. I don't think we will be able to do this in an if/else clause. It is ended to be.

* If we aren't storing the encrypted packet copy the packet and store
* encrypt
* If we are storing the encrypted packet copy the packet and store

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190?src=pr&el=h1) Report
> Merging [#190](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190?src=pr&el=desc) into [master](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/commit/fcda6244d06381a067ae85028b0eb5aa6ccaf51f&el=desc) will **increase** coverage by `0.00%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190?src=pr&el=tree)

```diff
@@           Coverage Diff           @@
##           master     #190   +/-   ##
=======================================
  Coverage   86.95%   86.95%           
=======================================
  Files          33       33           
  Lines        7698     7691    -7     
=======================================
- Hits         6694     6688    -6     
+ Misses       1004     1003    -1     
```


| [Impacted Files](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/source/Ice/Network.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9JY2UvTmV0d29yay5j) | `70.89% <0.00%> (ø)` | :arrow_up: |
| [src/source/Ice/TurnConnection.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9JY2UvVHVybkNvbm5lY3Rpb24uYw==) | `91.82% <0.00%> (ø)` | :arrow_up: |
| [src/source/Ice/ConnectionListener.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9JY2UvQ29ubmVjdGlvbkxpc3RlbmVyLmM=) | `90.50% <0.00%> (ø)` | :arrow_up: |
| [src/source/Ice/IceAgent.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9JY2UvSWNlQWdlbnQuYw==) | `93.81% <0.00%> (ø)` | :arrow_up: |
| [src/source/Ice/IceUtils.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9JY2UvSWNlVXRpbHMuYw==) | `96.80% <0.00%> (ø)` | :arrow_up: |
| [src/source/Ice/IceAgentStateMachine.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9JY2UvSWNlQWdlbnRTdGF0ZU1hY2hpbmUuYw==) | `82.97% <0.00%> (+0.30%)` | :arrow_up: |
| [src/source/PeerConnection/Rtp.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9QZWVyQ29ubmVjdGlvbi9SdHAuYw==) | `76.22% <0.00%> (+0.86%)` | :arrow_up: |
| [src/source/Ice/SocketConnection.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9JY2UvU29ja2V0Q29ubmVjdGlvbi5j) | `82.68% <0.00%> (ø)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190?src=pr&el=footer). Last update [b8116ed...cb632db](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/190?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Pull request merged**